libcxx/include/array

```diff
@@ -280,7 +280,8 @@ struct _LIBCPP_TEMPLATE_VIS array<_Tp, 0>
     typedef std::reverse_iterator<iterator>       reverse_iterator;
     typedef std::reverse_iterator<const_iterator> const_reverse_iterator;
 
-    typedef __conditional_t<is_const<_Tp>::value, const char, char> _CharType;
+    struct _EmptyAggregate { };
+    typedef __conditional_t<is_const<_Tp>::value, const _EmptyAggregate, _EmptyAggregate> _CharType;
```

Comment:
```suggestion
    using _EmptyType = __maybe_const<_Tp>::value, _EmptyAggregate>;
```
// #include <__type_traits/maybe_const.h> is needed too
The name was not great before, but now it looks really weird since there is no connection with `char` at all.

Reply:
I agree with renaming to `_EmptyType`, that was an oversight. However, I'd like to keep the `__maybe_const` out of this change, since it's really unrelated.